lib/utils.pm

```diff
@@ -1658,6 +1659,43 @@ sub script_output_retry {
     die("Waiting for Godot: $cmd") if $die;
 }
 
+
+=head2 validate_script_output_retry
+
+ validate_script_output_retry($cmd, $check, [retry => $retry], [delay => $delay], [timeout => $timeout]);
+
+Repeat command until validate_script_output succeeds or die. Return the output of the command on success.
+
+C<$retry> refers to the number of retries and defaults to C<10>.
+
+C<$delay> is the time between retries and defaults to C<30>.
+
+If the command doesn't succeed after C<$retry> retries,
+this function will die.
+
+Example:
+
+ validate_script_output_retry('ping -c1 -W1 machine', m/1 packets transmitted/, retry => 5, delay => 60);
+
+=cut
+sub validate_script_output_retry {
+    my ($cmd, $check) = splice(@_, 0, 2);
+    my %args = testapi::compat_args({retry => 10, delay => 30, timeout => 90}, [], @_);
+    my $retry = delete $args{retry};
+    my $delay = delete $args{delay};
+
+    my $exec = "timeout " . ($args{timeout} - 3) . " $cmd";
+    for (1 .. $retry) {
+        my $ret;
+        eval { $ret = validate_script_output($exec, $check, %args); };
+        return $ret if (defined($ret));
+        die($@) if ($@ =~ 'output not validating at');
```

Comment:
Where does this string `output not validating at` comes from? I only found this https://github.com/os-autoinst/os-autoinst/blob/master/testapi.pm#L1287 without `at`.

And I'm not sure about this. Because for me it looks like you only wanna retry on timeout. And for me `timeout` is actually an undefined state without some recover code.

Reply:
when the regex is not matched and `validate_script_output` failed, `$@` contains the whole error log `output not validating at /usr/lib/os-autoinst/testapi.pm line XYZ`  
But I will remove the `at`. 


Yes, as suggested by @asmorodskyi , it doesn't make sense to retry again and again if `validate_script_output` fails because the output doesn't match and not because of timeout. I don't know what's the best option... 
In my experience, in some cases `validate_script_output` failed because of unknown error (random network issue that returns !=0)  https://openqa.suse.de/tests/8189613#step/image_docker/318
Here the command fails and makes the test fail, not due to command timeout, but due to some random exit code. So, I will remove this condition.

Reply:
ok, but in that case, the return error is not `output not validating`... so It should be ok to keep this condition in the end...

Reply:
yes, these corner cases can happen. Thanks for the suggestion!